guard/src/commands/validate/cfn_reporter.rs

```diff
@@ -0,0 +1,187 @@
+                                Some(caps) => {
+                                    (caps["name"].to_string(), caps["rest"].replace("/", "."))
+                                },
+                                None =>
+                                    (format!("Rule {} Resource {} {}", each_failed_rule.context, idx, clause_idx), "".to_string())
+
+                            };
+                            resource_info.path = property_path;
+                            by_resource_name.entry(resource_name).or_insert(Vec::new()).push(resource_info);
+                        },
+
+                        _ => unreachable!()
+                    }
+                }
+            }
+            by_resource_name
+        } else { HashMap::new() };
+        let as_vec = passed_or_skipped.iter().map(|s| *s)
+            .collect::<Vec<&StatusContext>>();
+        let (skipped, passed): (Vec<&StatusContext>, Vec<&StatusContext>) = as_vec.iter()
+            .partition(|status| match status.status { // This uses the dereference deep trait of Rust
+                Some(Status::SKIP) => true,
+                _ => false
+            });
+        let skipped = skipped.iter().map(|s| s.context.clone()).collect::<HashSet<String>>();
+        let passed = passed.iter().map(|s| s.context.clone()).collect::<HashSet<String>>();
+        self.render.report(writer, self.rules_file_name, self.data_file_name, failed, passed, skipped, longest_rule_name)?;
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct SingleLineReporter {}
+
+impl super::common::GenericReporter for SingleLineReporter {
+    fn report(&self,
+              writer: &mut dyn Write,
+              rules_file_name: &str,
+              data_file_name: &str,
+              by_resource_name: HashMap<String, Vec<NameInfo<'_>>>,
+              passed: HashSet<String>,
+              skipped: HashSet<String>,
+              longest_rule_len: usize) -> crate::rules::Result<()> {
+
+        writeln!(writer, "Evaluation of rules {} for template {}, number of resource failures = {}",
+                 rules_file_name, data_file_name, by_resource_name.len())?;
+        if !by_resource_name.is_empty() {
+            writeln!(writer, "--");
+        }
+        //
+        // Agreed on text
+        // Resource [NewVolume2] property [Properties.Encrypted] in template [template.json] is not compliant with [sg.guard/aws_ec2_volume_checks] because provided value [false] does not match with expected value [true]. Error Message [[EC2-008] : EC2 volumes should be encrypted]
+        //
+        for (resource, info) in by_resource_name.iter() {
+            super::common::print_name_info(
+                writer, &info, longest_rule_len, rules_file_name, data_file_name,
+                |_, _, info| {
+                    Ok(format!("Resource [{}] traversed until [{}] for template [{}] wasn't compliant with [{}/{}] due to retrieval error. Error Message [{}]",
```

Comment:
The functions retrieval_error_message, unary_error_message, binary_error_message are abstracted out pretty nicely in the generic summary, but not here. Is there a reason for that? IMO, its just easier to read what is getting passed on to print_name_info.

Reply:
This is because the closures are capturing the `resource` variable from the loop. See

```rust
        for (resource, info) in by_resource_name.iter() {
            super::common::print_name_info(
                writer, &info, longest_rule_len, rules_file_name, data_file_name,
```

all other functionality that is common between these reporters is abstracted out into common functions. Some were made redundant as we will revisit these when appropriate for messaging changes